core/server/services/settings/index.js

```diff
@@ -12,11 +12,18 @@ module.exports = {
     },
 
     async reinit() {
+        const oldSettings = SettingsCache.getAll();
+
         SettingsCache.shutdown();
         const settingsCollection = await models.Settings.populateDefaults();
-        SettingsCache.init(settingsCollection);
+        const newSettings = SettingsCache.init(settingsCollection);
+
         for (const model of settingsCollection.models) {
-            model.emitChange(model.attributes.key + '.' + 'edited', {});
+            const key = model.attributes.key;
+
+            if (newSettings[key] !== oldSettings[key]) {
```

Comment:
This comparison is comparing object literals which will never be equal to each other.

I think this should be comparing the `value` of the settings!

Reply:
You're right. I fixed it and wrote unit tests for it.